engine/common/src/main/java/org/enso/common/ContextFactory.java

```diff
@@ -176,7 +176,7 @@ public Context build() {
     }
     var julLogLevel = Converter.toJavaLevel(logLevel);
     var logLevelName = julLogLevel.getName();
-    var inAOTMode = java.lang.Boolean.getBoolean("com.oracle.graalvm.isaot");
+    var inAOTMode = HostEnsoUtils.isAot();
     java.util.Map<String, String> engineOptions = null;
     if (runtimerServerKey != null) {
       if (!inAOTMode) {
```

Comment:
`inAOTMode` can be `true`, but `runtimeServerKey` can be `null`. Then `engineOptions` remain `null`.